hints/linux.sh

```diff
@@ -85,7 +85,7 @@ uname_minus_m="${uname_minus_m:-"$targetarch"}"
 
 # Check if we're about to use Intel's ICC compiler
 case "`${cc:-cc} -V 2>&1`" in
-*"Intel(R) C++ Compiler"*|*"Intel(R) C Compiler"*)
+*"Intel(R) C++ Compiler"*|*"Intel(R) C++ Intel(R) 64 Compiler"*|*"Intel(R) C Compiler"*|*"Intel(R) C Intel(R) 64 Compiler"*)
```

Comment:
Since we're treating all Intel compilers the same, regardless of C/C++ and 32/64 bitness, could this not simply be `*"Intel (R)C"*" Compiler"*`, similar to the `*" Sun "*"C"*` case just below?

Reply:
As I understand it Intel does not produce a separate compiler for 32 bit systems anymore.  The compiler for x86_64 can produce both 32 and 64 bit but both modes include the "64" in the version output:

```
❌ starscream% icc -m32 -V
Intel(R) C Intel(R) 64 Compiler for applications running on IA-32, Version 19.1.1.217 Build 20200306
Copyright (C) 1985-2020 Intel Corporation.  All rights reserved.
FOR NON-COMMERCIAL USE ONLY
```

I do agree your suggestion does seem more likely to catch variations on the output that might happen in the future, so I will test your suggestion and update this PR.

Reply:
I wasn't referring to the actual bitness of the compiled code, only what it was calling itself in the `-V` output.
All the strings that have been evidenced:
```
Intel(R) C++ Compiler
Intel(R) C Compiler
Intel(R) C++ Intel(R) 64 Compiler
Intel(R) C Intel(R) 64 Compiler
```
can be matched by the pattern
```
*"Intel(R) C"*" Compiler"*
```

Reply:
Thank you for clarifying.  I addressed two points, as I think you were asking `-V` on a 32 bit system, which is no longer available, and also `-V` when running in 32 bit mode on x86_64 (which does actually print a different version string, although it still matches this case).